frontend/src/components/MyGroupsProvider.js

```diff
@@ -27,7 +32,7 @@ export default function MyGroupsProvider({ children, authenticated }) {
   }, [authenticated]);
```

Comment:
Couple of questions, hope you don't mind: Wouldn't starting as authenticated mean that the data is not refetched when a user _becomes_ authenticated? Seems weird, also, to assume a user is authenticated by default. 

Also couldn't you just use the AppLoadingContext and throw the whole app into a loading state instead of tracking loading separately here?